tests/unit/api/objectReplicationMD.js

```diff
@@ -0,0 +1,304 @@
+const assert = require('assert');
+const async = require('async');
+
+const { cleanup, DummyRequestLogger, makeAuthInfo, TaggingConfigTester } =
+    require('../helpers');
+const { metadata } = require('../../../lib/metadata/in_memory/metadata');
+const DummyRequest = require('../DummyRequest');
+const { bucketPut } = require('../../../lib/api/bucketPut');
+const objectDelete = require('../../../lib/api/objectDelete');
```

Comment:
I've mocked the APIs for bucketPut, bucketPutVersioning, and bucketPutReplication. The remaining APIs follow the paths to apiUtil replication methods added by this PR, so I've kept them. We could also remove them and test the apiUtil methods directly, but I'm thinking this way might be a bit more thorough.